addon/mixins/resize-text.js

```diff
@@ -23,24 +41,34 @@ export default Mixin.create({
   },
 
   scaleFont() {
-    const style = getComputedStyle(this.element);
-    const minSize = this.get('minSize');
-    const maxSize = this.get('maxSize');
-    const container = this.get('containerElement');
-    let elementWidth;
-    if (container) {
-      elementWidth = container.clientWidth;
-    } else {
-      elementWidth = this.element.clientWidth;
-    }
+    const container = this.get('containerElement') || this.element;
+    const style = getComputedStyle(container);
+    const currentFontSize = extractPixels(style.fontSize);
+
+    // if shrink or grow are set to false, use the currentFontSize
+    const { shrink, grow } = this.getProperties('shrink', 'grow');
+    const minSize = !shrink ? currentFontSize : this.get('minSize');
+    const maxSize = !grow ? currentFontSize : this.get('maxSize');
+
+    // the computed style.width is the given width, w/o padding, which is what we want
+    let elementWidth = extractPixels(style.width) || container.clientWidth;
+    // remove the padding, which is the actual inner/renderable width
+    elementWidth -= extractPixels(style.paddingLeft);
+    elementWidth -= extractPixels(style.paddingRight);
+
     let fontSize = this.get('textMeasurer')
-        .fitTextSize(this.element.innerText, elementWidth, style.fontStyle + ' 14px ' + style.fontFamily);
+        .fitTextSize(this.element.innerText, elementWidth,
+          `${style.fontStyle} ${style.fontSize} ${style.fontFamily}`);
```

Comment:
this won't do. The font size is ignored by the fitTextSize method. please see https://github.com/cibernox/ember-text-measurer
```
fitTextSize(string, width, font = null) will return the font size with which the given text will fully fit in the available width with the given (option) font information.
textMeasurer.fitTextSize('foobar this is too long', 200, 'normal 24px Times'); // 22px
textMeasurer.fitTextSize('foobar this is too long', 200, 'normal 24px Helvetica'); // 21px
textMeasurer.fitTextSize('foobar this is too long', 200, 'normal 24px Georgia'); // 20px
Note: The font size passed to this method is irrelevant, the important part is the font style and family.
```

Reply:
:+1: